clients/src/main/java/org/apache/kafka/common/requests/AddPartitionsToTxnResponse.java

```diff
@@ -99,6 +112,7 @@ public void maybeSetThrottleTimeMs(int throttleTimeMs) {
         data.setThrottleTimeMs(throttleTimeMs);
     }
 
+    // Only used for versions < 4
```

Comment:
Do we need to continue exposing some of these version-specific methods? We sort of intend the request/response objects to abstract a lot of the version-specific details.

Reply:
I'm not sure how to get around some of these since the return type is different.

Reply:
As mentioned elsewhere, the old API is a special case of the new API where there just happens to be a single transactionalId. I think if we take that perspective, then we can make the methods general and hide the version details.

Reply:
I guess if we want to deal with a map or larger data structure that has only one object we can. I think the tricky thing is knowing when we expect just one response how to use it correctly. That may require further refactoring but it is probably doable.